### DISCUSSION TITLE
[GLFW][OPENGL3] Runtime Error when setup 2 glfw window with dear imgui frame on it

### DISCUSSION THREAD

**Discussion by binaryfox0** [ocornut/imgui#6291] [New Users Build/Link/Run issues ONLY!]
I am a Junior Developer, and use ImGui for simple text editor. When I do a save dialog from opengl in a new window of glfw (not in main window). Compile isnt have any error but when run it runtime error appear

![image](https://user-images.githubusercontent.com/101315124/229297463-ba4b7796-43d7-4b15-b535-c928bdaca72c.png)

Here is my code
`// Main window (first window)
window[1] = glfwCreateWindow(1280, 720, "Dear ImGui GLFW+OpenGL3 example", NULL, NULL);
	if (window == NULL)
		return 1;
	glfwMakeContextCurrent(window[1]);
	glfwSwapInterval(1); // Enable vsync
	
  // Dialog call
	if(isDialogCall == true)
	{
    If dialog is call it will be create second window
		window[2] = glfwCreateWindow(1280, 720, "File Explorer", NULL, NULL);
	}

	// Setup Dear ImGui context
	IMGUI_CHECKVERSION();
	ImGui::CreateContext();
	ImGuiIO& io = ImGui::GetIO(); (void)io;
	io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;     // Enable Keyboard Controls
	io.ConfigFlags |= ImGuiConfigFlags_NavEnableGamepad;      // Enable Gamepad Controls

	// Setup Dear ImGui style
	ImGui::StyleColorsDark();
	//ImGui::StyleColorsLight();

	// Setup Platform/Renderer backends
	ImGui_ImplGlfw_InitForOpenGL(window[1], true);
	ImGui_ImplOpenGL3_Init(glsl_version);

	if (isDialogCall == true)
	{
		ImGui_ImplGlfw_InitForOpenGL(window[1], true);
		ImGui_ImplOpenGL3_Init(glsl_version);
	}

	// Load Fonts
	// (there is a default font, this is only if you want to change it. see extra_fonts/README.txt for more details)
	ImVec4 clear_col = ImColor(114, 144, 154);

        ... // Text Editor Code Here

	// Main loop
	while (!glfwWindowShouldClose(window[1]))
	{
		glfwPollEvents();

		// Start the Dear ImGui frame
		ImGui_ImplOpenGL3_NewFrame();
		ImGui_ImplGlfw_NewFrame();
		ImGui::NewFrame();

		ImGui::SetNextWindowPos(ImVec2(0.0f, 0.0f));

// Main frame of first window	ImGui::SetNextWindowSize(ImGui::GetIO().DisplaySize);
		auto cpos = editor.GetCursorPosition();
		ImGui::Begin("Text Editor Demo", nullptr, ImGuiWindowFlags_HorizontalScrollbar | ImGuiWindowFlags_MenuBar | ImGuiWindowFlags_NoTitleBar);
		if (ImGui::BeginMenuBar())
		{
			if (ImGui::BeginMenu("File"))
			{
				if (ImGui::MenuItem("Save"))
				{
					// Editor code here
				}
				if (ImGui::MenuItem("Quit", "Alt-F4"))
					break;
				ImGui::EndMenu();
			}
			if (ImGui::BeginMenu("Edit"))
			{
				//Editor code here
			}

			//Editor code here
		}

		//Editor code here

		ImGui::End();

		// Rendering
		ImGui::Render();
		int display_w, display_h;
		glfwGetFramebufferSize(window[1], &display_w, &display_h);
		glViewport(0, 0, display_w, display_h);
		glClearColor(ImVec4(0.45f, 0.55f, 0.60f, 1.00f).x * ImVec4(0.45f, 0.55f, 0.60f, 1.00f).w, ImVec4(0.45f, 0.55f, 0.60f, 1.00f).y * ImVec4(0.45f, 0.55f, 0.60f, 1.00f).w, ImVec4(0.45f, 0.55f, 0.60f, 1.00f).z * ImVec4(0.45f, 0.55f, 0.60f, 1.00f).w, ImVec4(0.45f, 0.55f, 0.60f, 1.00f).w);
		glClear(GL_COLOR_BUFFER_BIT);
		ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
		
		glfwSwapBuffers(window[1]);

		if (isDialogCall)
		{
       // Create frame in second window
			ImGui::Begin("File Explorer", nullptr, ImGuiWindowFlags_HorizontalScrollbar);
			ImGui::End();
			ImGui::Render();
			ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());

			glfwSwapBuffers(window[1]);
		}
	}`

**Comment by ocornut** [comment]
Try to guess what the assert means and press Retry to find the code location and guess what it means.
